c/parallel/api_primitives.h

```diff
@@ -49,6 +49,17 @@ class NThreads {
       size_t n0 = num_threads_in_pool();
       value = (nth == 0 || nth >= n0) ? n0 : nth;
     }
+    // Calculate number of threads from the number of iterations
+    // and the minimum number of iterations per thread.
+    explicit NThreads(size_t niters, size_t min_iters_per_thread) {
```

Comment:
I don't think that's a good API. This will lead to code such as `NThreads(n, 10)`, and that will totally confuse the reader... Besides, I'm not even sure that the NThreads class should know anything about any iterations

Reply:
For my purpose it doesn't really matter, so I've converted the constructor into a separate function.